Userland/Libraries/LibJS/Parser.cpp

```diff
@@ -409,7 +409,7 @@ Parser::ParserState::ParserState(Lexer l, Program::Type program_type)
 }
 
 Parser::Parser(Lexer lexer, Program::Type program_type, Optional<EvalInitialState> initial_state_for_eval)
-    : m_source_code(SourceCode::create(lexer.filename(), lexer.source()))
+    : m_source_code(SourceCode::create((String::from_deprecated_string(lexer.filename()).value()), String::from_deprecated_string(lexer.source()).value()))
```

Comment:
Unsure what makes the most sense here. There's no `vm` reference available within Parser so `TRY_OR_THROW_OOM` is not easily usable.